airbyte-webapp/src/packages/cloud/services/WithAnalyticsContext.tsx

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from "react";
+
+import { useAnalyticsCtx } from "hooks/useAnalytics";
+import { useCurrentWorkspace } from "hooks/services/useWorkspace";
+
+import { useAuthService } from "./auth/AuthService";
+
+const WithAnalyticsContext: React.FC = ({ children }) => {
+  const ctx = useAnalyticsCtx();
+
+  const { workspaceId, customerId } = useCurrentWorkspace();
+  const { user } = useAuthService();
+
+  useEffect(() => {
```

Comment:
I believe it's better to add it as a separate hook, so it could be used as. 

```
useProvideAnalyticsProp({workspaceId, customerId} // or some other better name
```

so it will add and remove from context where necessary.